File: js/timeConversion.js
function timeConversion(s) {
  // Write your code here
  let final = "";
  if (s[0] + s[1] == "12" && s[8] + s[9] == "AM") {
    let first = s[0] + s[1];
    first = "00";
    final = first + s[2] + s[3] + s[4] + s[5] + s[6] + s[7];
  } else if (s[0] + s[1] >= 1 && s[0] + s[1] <= 12 && s[8] + s[9] == "AM") {
    final = s[0] + s[1] + s[2] + s[3] + s[4] + s[5] + s[6] + s[7];
  } else if (s[0] + s[1] >= 1 && s[0] + s[1] <= 11 && s[8] + s[9] == "PM") {
    let sum = parseInt(s[0] + s[1]) + 12;
    final = sum + s[2] + s[3] + s[4] + s[5] + s[6] + s[7];
  }

  return final;
}

// 12:00 am -> 00:00
// 01:00 - 12:00 am -> keep just remove am
// 01:00 - 11:00 pm -> sum to 12
timeConversion("12:00:00AM");
timeConversion("01:00:00AM");
timeConversion("07:05:45PM");

File: go/basic/timeConversion.go
package basic

import (
	"strconv"
)

func TimeConversion(s string) string {
	// Write your code here
	a, _ := strconv.Atoi(string(s[0]) + string(s[1]))
	if a == 12 && string(s[8])+string(s[9]) == "AM" {
		s = "00" + string(s[2]) + string(s[3]) + string(s[4]) + string(s[5]) + string(s[6]) + string(s[7])
	} else if a >= 1 && a <= 12 && string(s[8])+string(s[9]) == "AM" {
		s = string(s[0]) + string(s[1]) + string(s[2]) + string(s[3]) + string(s[4]) + string(s[5]) + string(s[6]) + string(s[7])
	} else if a >= 1 && a <= 11 && string(s[8])+string(s[9]) == "PM" {
		sum := a + 12
		str := strconv.Itoa(sum)
		s = str + string(s[2]) + string(s[3]) + string(s[4]) + string(s[5]) + string(s[6]) + string(s[7])
	} else if a == 12 && string(s[8])+string(s[9]) == "PM" {
		s = string(s[0]) + string(s[1]) + string(s[2]) + string(s[3]) + string(s[4]) + string(s[5]) + string(s[6]) + string(s[7])
	}
	return s
}
